impeller/compiler/reflector.cc

```diff
@@ -571,6 +573,7 @@ std::vector<StructMember> Reflector::ReadStructMembers(
     const auto& member = compiler_->get_type(struct_type.member_types[i]);
     const auto struct_member_offset =
         compiler_->type_struct_member_offset(struct_type, i);
+    auto array_elements = std::max(1u, GetArrayElements(member));
```

Comment:
Why is this capped at 1?

Reply:
Oh, nevermind.

I'm trying to figure out why unbounded glsl arrays are getting set to 1 and thought this might have been the problem.

Reply:
Ok, it is the problem. `GetArrayElements` is returning zero for dynamically sized arrays, and we're just saying they always have a length of 1, which is working for reasons I don't understand but seems wrong.

Reply:
This was for the metadata of static sized arrays within uniform structs. Zero length arrays members aren't legal in C++ -- the uniform mapping ends up working out because everything still gets spaced out properly. Maybe it would be less confusing to just drop the field in this situation.